### ISSUE TITLE
Query params in location breaks routing

### ISSUE THREAD

**Issue by mkotlikov** [ktorio/ktor#234]
Works: 
`http://localhost:12345/foo/123`
`http://localhost:12345/foo/bar?fus=a&roh=b`
Broken(`bar` is processed as `{id}` and triggers a `NumberFormatException`):
`http://localhost:12345/foo/bar`
`http://localhost:12345/foo/bar?fus=a`
`http://localhost:12345/foo/bar?roh=b`

```
import org.jetbrains.ktor.application.install
import org.jetbrains.ktor.host.embeddedServer
import org.jetbrains.ktor.locations.Locations
import org.jetbrains.ktor.locations.get
import org.jetbrains.ktor.locations.location
import org.jetbrains.ktor.netty.Netty
import org.jetbrains.ktor.response.respond
import org.jetbrains.ktor.routing.Routing
import org.jetbrains.ktor.routing.get

@location("/foo/{id}")
data class FooWithIdLocation(val id: Long)

@location("/foo/bar")
data class FooBarLocation(val fus: String = "", val roh: String = "")

fun main(args: Array<String>) {
    embeddedServer(Netty, 12345) {
        install(Locations)
        install(Routing) {
            get<FooBarLocation>{
                call.respond("${it.fus} | ${it.roh}")
            }

            get<FooWithIdLocation>{
                call.respond("${it.id}")
            }
        }
    }.start(wait = true)
}
```

**Comment by mkotlikov** [comment]
The workaround is not to use a location for the static route.
Works:
`http://localhost:12345/foo/123`
`http://localhost:12345/foo/bar?fus=a&roh=b`
`http://localhost:12345/foo/bar`
`http://localhost:12345/foo/bar?fus=a`
`http://localhost:12345/foo/bar?roh=b`
```
import org.jetbrains.ktor.application.install
import org.jetbrains.ktor.host.embeddedServer
import org.jetbrains.ktor.locations.Locations
import org.jetbrains.ktor.locations.get
import org.jetbrains.ktor.locations.location
import org.jetbrains.ktor.netty.Netty
import org.jetbrains.ktor.response.respond
import org.jetbrains.ktor.routing.Routing
import org.jetbrains.ktor.routing.get

@location("/foo/{id}")
data class FooWithIdLocation(val id: Long)

fun main(args: Array<String>) {
    embeddedServer(Netty, 12345) {
        install(Locations)
        install(Routing) {
            get("/foo/bar"){
                val fus = call.request.queryParameters.get("fus") ?: ""
                val roh = call.request.queryParameters.get("roh") ?: ""
                call.respond("$fus | $roh")
            }

            get<FooWithIdLocation>{
                call.respond("${it.id}")
            }
        }
    }.start(wait = true)
}
```

**Comment by orangy** [comment]
This is fixed in `0.9`
